plugin/pkg/shoot/validator/admission.go

```diff
@@ -1113,7 +1115,21 @@ func getDefaultMachineImage(machineImages []core.MachineImage, imageName string,
 			return nil, fmt.Errorf("image name %q is not supported", imageName)
 		}
 	} else {
-		defaultImage = &machineImages[0]
+		//select the first image which support the required architecture type
+		for _, machineImage := range machineImages {
+			for _, verison := range machineImage.Versions {
+				if slices.Contains(verison.Architectures, *arch) {
+					defaultImage = &machineImage
+					break
+				}
+			}
+			if defaultImage != nil {
+				break
+			}
+		}
+		if defaultImage == nil {
+			return nil, fmt.Errorf("image name %q is not supported", imageName)
```

Comment:
In which situations can this happen? Probably only when there is no machine image version with the respective architecture, right? Can you enhance the error message?